applications/StructuralMechanicsApplication/custom_elements/total_lagrangian.cpp

```diff
@@ -107,11 +107,11 @@ namespace Kratos
         // If strain has to be computed inside of the constitutive law with PK2
         Values.SetStrainVector(this_constitutive_variables.StrainVector); //this is the input  parameter
 
-        // Contribution to external forces
-        const Vector body_force = this->GetBodyForce();
-        
         for ( unsigned int point_number = 0; point_number < integration_points.size(); point_number++ )
         {
+            // Contribution to external forces
+            const Vector body_force = this->GetBodyForce(integration_points[point_number].Coordinates());
```

Comment:
pass point_number instead, or even better, pass directly the array of Ns taken from the kinematic variables